app/main.py

```diff
@@ -1,3 +1,19 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> Callable:
+    cached = []
+
+    def wrapper(*args, **kwargs) -> int:
+        cache_tuple = (*args, frozenset(kwargs.items()))
```

Comment:
Why do we freeze ordering from kwargs.items()?

Reply:
I wanted to prevent accidental deletion of data in future but now I see that there is no need to use it in that place.

Reply:
> Why do we freeze ordering from kwargs.items()?

That's because if we use regular set, the hash will depend on the order of kwargs, on the other side if we use frozenset the order of elemenent won't matter and won't influence on hash. So we will be able to check or retrieve information from our cached list correctly.

Reply:
> Okay but how the order of the cached elements make an impact here?

When caching results based on the arguments (both positional and keyword arguments) of a function, you need a way to generate a unique and consistent hash for these arguments so that you can efficiently retrieve cached results. Using a simple set to store the arguments wouldn't be sufficient because sets are unordered collections, and their hash values could change based on the order of elements. This means that if you were to call the same function with the same arguments in a different order, you might not be able to retrieve the cached result, even though the actual computation is the same.

By using a frozenset, you're creating an immutable and hashable object that represents the set of keyword arguments, without being affected by their order. This ensures that no matter in which order the keyword arguments were passed, the resulting frozenset will have the same hash value. As a result, you can reliably use this hash value as part of your caching mechanism to retrieve the cached result associated with those specific keyword arguments, regardless of their order.

In conclusion, using a frozenset eliminates any potential inconsistency caused by the order of the keyword arguments and ensures that the caching mechanism works as expected in all scenarios.

Reply:
Okay, maybe I don't quiet understand here, because you replied me a doc for frozenset 
We're caching this key
![image](https://github.com/mate-academy/py-cache-decorator/assets/73850362/b2eec04f-ac5b-4c21-b891-2f88c25be2ce)
But should we really do so? Also, if applying frozenset to dict_items object makes sense in terms of obtaining the correct and cached key, can you give me an example?